### PR TITLE
NET-187: Event stream for storage node assignments

### PR DESCRIPTION
**Pull request by teogeb**
Sends an event to a separate event stream when a streams is added to a storage node (or removed from a storage node).

All events are sent to the same stream (${EE_address}/storage-node-assignments`), and the stream is created when the application starts, if needed.

### CODE REVIEW

File: src/groovy/com/unifina/domain/EthereumAddress.groovy [hunk 1]
```diff
@@ -20,4 +22,11 @@ class EthereumAddress {
 	public String toString() {
 		return value
 	}
+
+	static EthereumAddress fromPrivateKey(String privateKey) {
+		BigInteger pk = new BigInteger(privateKey, 16)
+		ECKey key = ECKey.fromPrivate(pk)
+		String publicKey = "0x" + Hex.encodeHexString(key.getAddress())
```
Comment (kare) [inline src/groovy/com/unifina/domain/EthereumAddress.groovy:29]: Dependencies for commons codec and org.ethereum are going to be dropped. Use Web3j methods to accomplish same thing:
```
final String address = Keys.getAddress(publicKey);
final String publicKey = Numeric.prependHexPrefix(address);
```

Comment (teogeb) [reply src/groovy/com/unifina/domain/EthereumAddress.groovy]: Fixed

File: src/groovy/com/unifina/domain/EthereumAddress.groovy [hunk 2]
```diff
@@ -24,9 +26,12 @@ class EthereumAddress {
 	}
 
 	static EthereumAddress fromPrivateKey(String privateKey) {
+		if (privateKey.startsWith(PREFIX)) {
+			privateKey = privateKey.substring(PREFIX.length())
```
Comment (kare) [inline src/groovy/com/unifina/domain/EthereumAddress.groovy:30]: FWIW, there's `Numeric.cleanHexPrefix()`

File: src/java/com/unifina/utils/ApplicationConfig.groovy [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.unifina.utils
+
+import grails.util.Holders
+import grails.compiler.GrailsCompileStatic
+import groovy.transform.EqualsAndHashCode
+
+@GrailsCompileStatic
+class ApplicationConfig {
```
Comment (kare) [inline src/java/com/unifina/utils/ApplicationConfig.groovy:8]: 👍

**Review by kare — APPROVED** [review]
One comment about the general existence of `Numeric.cleanHexPrefix()`. LGTM

**Pull request merged**